### PR TITLE
Upgrade Vitest and related testing deps

### PR DESCRIPTION
**Pull request by cprecioso**
## Description

- Blocking #3571 
- Resolves https://github.com/wasp-lang/wasp/issues/2319

We've been using Vitest 1 in Wasp Testing, which only supports up to Vite 4; but we're using Vite 7. That was incorrect.

One of the symptoms is that `defineConfig` from `vitest` is not up-to-date with the one from `vite`, so the Tailwind upgrade at #3571 was blocked due to incompatible plugin types.

## Type of change

- [ ] **🔧 Just code/docs improvement** 
- [x] **🐞 Bug fix** 
- [ ] **🚀 New/improved feature** 
- [x] **💥 Breaking change** 

## Checklist

- [x] I tested my change in a Wasp app to verify that it works as intended.

- 🧪 Tests and apps:

  - [ ] I added **unit tests** for my change. 
  - [ ] _(if you fixed a bug)_ I added a **regression test** for the bug I fixed. 
  - [ ] _(if you added/updated a feature)_ I added/updated **e2e tests** in `examples/kitchen-sink/e2e-tests`.
  - [ ] _(if you added/updated a feature)_ I updated the **starter templates** in `waspc/data/Cli/templates`, as needed.
  - [x] _(if you added/updated a feature)_ I updated the **example apps** in `examples/`, as needed.
    - [ ] _(if you updated `examples/tutorials`)_ I updated the tutorial in the docs (and vice versa).

- 📜 Documentation:

  - [ ] _(if you added/updated a feature)_ I **added/updated the documentation** in `web/docs/`.

- 🆕 Changelog: _(if change is more than just code/docs improvement)_
  - [x] I updated `waspc/ChangeLog.md` with a **user-friendly** description of the change.
  - [x] _(if you did a breaking change)_ I added a step to the current **migration guide** in `web/docs/migration-guides/`.
  - [x] I **bumped the `version`** in `waspc/waspc.cabal` to reflect the changes I introduced.

### CODE REVIEW

File: examples/kitchen-sink/src/features/operations/components/Todo.test.tsx [hunk 1]
```diff
@@ -69,6 +70,7 @@ test("handles multiple mock data sources", async () => {
   mockQuery(getMe, mockUser);
   mockQuery(getDate, new Date());
   mockQuery(getTasks, mockTasks);
+  mockQuery(getNumTasks, mockTasks.length);
```
Comment (cprecioso) [inline examples/kitchen-sink/src/features/operations/components/Todo.test.tsx:73]: This mock was missing

File: waspc/data/Cli/starters/skeleton/src/vite-env.d.ts [hunk 2]
```diff
@@ -1,7 +1,3 @@
 /// <reference types="vite/client" />
-
-// This is needed to properly support Vitest testing with jest-dom matchers.
-// Types for jest-dom are not recognized automatically and Typescript complains
-// about missing types e.g. when using `toBeInTheDocument` and other matchers.
-// Reference: https://github.com/testing-library/jest-dom/issues/546#issuecomment-1889884843
-import "@testing-library/jest-dom";
```
Comment (cprecioso) [inline waspc/data/Cli/starters/skeleton/src/vite-env.d.ts:7]: We don't really need this note, this is just the normal way to install this library as per its documentation

File: web/docs/migration-guides/migrate-from-0-20-to-0-21.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+}
+```
+
+### 2. Upgrade Vitest tests to v4
+
+**If you don't have test files in your project, you can skip this step.**
+
+We upgraded our testing support from Vitest v1 to Vitest v4. We expect most tests to work without any changes, but if you have custom test setups or configurations, please upgrade your tests by following the migration guides from Vitest, in order:
+
+:::note
```
Comment (infomiho) [inline web/docs/migration-guides/migrate-from-0-20-to-0-21.md]: Great note!

File: waspc/data/Cli/starters/skeleton/src/vite-env.d.ts (hunk 2 above)
Comment (sodic) [reply waspc/data/Cli/starters/skeleton/src/vite-env.d.ts:7]: What about the new thing `vitest/globals`. Is that needed for Vitest 4?

Also, will we be able to kick all of this out once we move Vitest to userland?

File: web/docs/migration-guides/migrate-from-0-20-to-0-21.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+Most of the breaking changes are related to internal configuration, edge cases, and very advanced usage, so we recommend first to try running your tests after bumping the Wasp version, and only read through the migration guides if you encounter issues.
+:::
```
Comment (sodic) [inline web/docs/migration-guides/migrate-from-0-20-to-0-21.md]: I would move this note out of the callout and into the main text:
- The message is very similar to what we said in the paragraph above, so we should be able to easily merge them.
- It is a part of the main flow, not an extra note.
- Callouts kinda break the flow (especially when inserted right before a numbered list).

File: waspc/data/Cli/starters/skeleton/src/vite-env.d.ts (hunk 2 above)
Comment (cprecioso) [reply waspc/data/Cli/starters/skeleton/src/vite-env.d.ts:7]: > What about the new thing vitest/globals. Is that needed for Vitest 4?

Not v4-specific, but we should have added it before, since it is recommended when you enable the `globals: true` option (required for `@testing-library/react`): https://vitest.dev/config/globals.html#globals

> Also, will we be able to kick all of this out once we move Vitest to userland?

Mmmm tbh I haven't thought about it, I think it will depend on our specific implementation. But we're not better off or worse off with this change so I'm not especially worried.

File: web/docs/migration-guides/migrate-from-0-20-to-0-21.md (hunk 4 above)
Comment (cprecioso) [reply web/docs/migration-guides/migrate-from-0-20-to-0-21.md]: Yep, I evaluated that already, but I think that people will see a bullet list and skip directly to it, that's why I wanted to add an attention-grabbing callout. But let me re-construct this, I think I can do away with it.
